### PR TITLE
Python: preprend 'Semantic-Kernel' to User-Agent, and add version key-value pair to headers

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

The SK Python custom headers to track the USER_AGENT and the Python version were getting stuck into the default headers under the USER_AGENT key. Instead, they need to be represented like:

```
    {
        "UserAgent": HTTP_USER_AGENT,
        "Semantic-Kernel-Version": f"python-{version_info}",
    }
```

Trying to extract the Python version from telemetry data was not working as expected.

### Description

Use the APP_INFO dict as-is, and stick the key-value pairs into the default_headers.

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/10cb725e7b97d917a2685cedf12fc696f0a5fe34/README.md">Py3.8 Test Coverage</a><details><summary>Python 3.8 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel/connectors/ai/open_ai/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/10cb725e7b97d917a2685cedf12fc696f0a5fe34/python/semantic_kernel/connectors/ai/open_ai/services/azure_config_base.py">azure_config_base.py</a></td><td>34</td><td>2</td><td>94%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/10cb725e7b97d917a2685cedf12fc696f0a5fe34/python/semantic_kernel/connectors/ai/open_ai/services/azure_config_base.py#L80">80</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/10cb725e7b97d917a2685cedf12fc696f0a5fe34/python/semantic_kernel/connectors/ai/open_ai/services/azure_config_base.py#L96">96</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/10cb725e7b97d917a2685cedf12fc696f0a5fe34/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_config_base.py">open_ai_config_base.py</a></td><td>32</td><td>1</td><td>97%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/10cb725e7b97d917a2685cedf12fc696f0a5fe34/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_config_base.py#L56">56</a></td></tr><tr><td><b>TOTAL</b></td><td><b>5382</b></td><td><b>1008</b></td><td><b>81%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.8 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1205 | 11 :zzz: | 0 :x: | 0 :fire: | 25.112s :stopwatch: |

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_config_base.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def __init__(
         # Merge APP_INFO into the headers if it exists
         merged_headers = default_headers.copy() if default_headers else {}
         if APP_INFO:
-            merged_headers[USER_AGENT] = json.dumps(APP_INFO)
+            merged_headers.update(APP_INFO)
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/ai/open_ai/services/open_ai_config_base.py:52]: We need to preprend `Semantic-Kernel` to the current value of the User-Agent.

**Review by eavanvalkenburg — APPROVED** [review]
LGTM

**Review by juliomenendez — APPROVED** [review]
LGTM

**Pull request merged**